### PR TITLE
[WIP] Battling , Game Core and Game Progress

### PR DESCRIPTION
**Pull request by til61**
This PR is intended to solve:
1. the basics of battling, AKA distinguish player's monster with enemy monsters.
2. the collection of player's monsters, the "party".
3. explore using bevy resource to keep track and update game progress

### CODE REVIEW

File: src/backgrounds.rs [hunk 1]
```diff
@@ -212,4 +214,27 @@ pub(crate) fn expand_map(
             draw_chunk!(new_chunk, commands, map_atlas_handle);
         }
     }
-}
\ No newline at end of file
+}
+
+macro_rules! set_seed_ {
```
Comment (Elsklivet) [inline src/backgrounds.rs:219]: Changes to `backgrounds.rs` will probably conflict with #42, should this PR take priority?

File: src/battle.rs [hunk 2]
```diff
@@ -169,18 +171,18 @@ pub(crate) fn setup_battle(mut commands: Commands,
...
 pub(crate) fn setup_battle_stats(mut commands: Commands, 
 	asset_server: Res<AssetServer>,
-	mut player: Query<&mut Player>,
-	mut monster_query: Query<(&mut MonsterBundle)>) 
+	mut my_monster: Query<&mut MonsterBundle, (With<Fighting>, Without<Enemy>)>,
+	mut enemy: Query<&mut MonsterBundle, With<Enemy>>) 
 {
-	if player.is_empty() {
+	if my_monster.is_empty() {
 		error!("No player found!");
```
Comment (Elsklivet) [inline src/battle.rs:178]: It may be more descriptive to say, like, "no player monster found", but this is a nitpick

File: src/battle.rs [hunk 3]
```diff
@@ -505,6 +522,52 @@ pub (crate) fn attack_button_handler (
     }
 }
 
+fn calculate_damage(player: &MonsterBundle, player_action: usize,
```
Comment (Elsklivet) [inline src/battle.rs:525]: Again a naming nitpick, but I dislike calling the player's monster a "player" when it's not. I expected that a player's monster might drop to 0 HP without the player losing instantly, as they have other monsters. If this isn't the case, we have to make that more clear to players.

File: src/battle.rs [hunk 4]
```diff
@@ -505,6 +522,52 @@ pub (crate) fn attack_button_handler (
...
+        } else {
+            // calculate crit chance and apply crit damage
+            let mut crit_chance = player.stg.crt - enemy.def.crt_res;
+            let crit = rand::thread_rng().gen_range(0..=100);
+            if crit <= crit_chance {
+                result.0 *= player.stg.crt_dmg;
+            }
+        }
+    }
+    // same for enemy
```
Comment (Elsklivet) [inline src/battle.rs:552]: Could these be refactored into macros to clean up and practice DRY?

File: src/world.rs [hunk 5]
```diff
@@ -48,64 +48,50 @@ impl WorldMap{
...
+        self.get_chunk(x, y - 1)
     }
 
 }
     
 pub(crate) fn logical_to_rendering(x: isize, y: isize) -> (f32, f32){
     (x as f32 * WIN_W, y as f32 * WIN_H)
+}
+
+pub(crate) struct GameProgress{
```
Comment (Elsklivet) [inline src/world.rs:73]: I like this struct, but why underscores? Just to silence warnings or? The fields are already implicitly private, and the `_` naming convention is annoying to type, so I'd refactor `_` out unless it's here for warning suppression (but even in that case, we could use a temporary `#[allow(unused)]`)

File: src/backgrounds.rs (hunk 1 above)
Comment (til61) [reply src/backgrounds.rs:219]: This change is experimental and minimal, let main take priority.

File: src/world.rs (hunk 5 above)
Comment (til61) [reply src/world.rs:73]: silence warnings, and to show explicitly that this is not being used. In the spirit of commit often and commit little, function of game progress will not be introduced by this PR.

File: src/battle.rs (hunk 4 above)
Comment (til61) [reply src/battle.rs:594]: Feel free to try. I did not look into these.

File: src/battle.rs (hunk 3 above)
Comment (til61) [reply src/battle.rs]: Very good point. Open a issue so we can resolve this when we introduce monster bags.

**Pull request merged**